lib/rate_map.c

```diff
@@ -203,3 +210,95 @@ rate_map_shift_by_mass(rate_map_t *self, double pos, double mass)
     double result_mass = rate_map_position_to_mass(self, pos) + mass;
     return rate_map_mass_to_position(self, result_mass);
 }
+
+#ifdef _Static_assert
+_Static_assert(FLT_RADIX == 2, "Base 2 floating point types required");
```

Comment:
Looks like _Static_assert is only in C11, which we'll probably never use. On the one hand this is reasonably safe assumption so maybe we don't need it, but I guess we could have 
```
#if FLT_RADIX != 2
    abort()
#endif 
```
or similar?

Reply:
I got some interesting details on this at https://stackoverflow.com/questions/3385515/static-assert-in-c

_Static_assert has been in gcc since version 4.6 and it sounds like it shows up in clang and even VS. It just wasn't standardized until C11. I'm estimating that few if any users are going to compile with a compiler that doesn't have it. And if they don't then this static assert won't happen. But they probably are using IEEE base 2 floats anyway.

There are many ways to re-implement a static assert.

Reply:
@jeromekelleher @castedo For tskit we do static asserts with `#error` as in https://github.com/tskit-dev/tskit/blob/master/c/tskit/core.h#L74

Reply:
Yes, thinking about it again - given it's such an obscure and unlikely thing we're guarding against, it seems a bit perverse to depend on something else that we have to test existence for, just to check it. I.e., *if* we're on a platform where FLT_RADIX != 2, then I think there's a very good chance that ``_Static_assert`` doesn't exist either, therefore rendering the exercise pointless.

Reply:
thx @benjeffery, that is a far better. It should have occurred to me that nothing needs to get compiled, it's entirely a check that the preprocessor can handle with just `#error`.